spec/integration/inventory2_spec.rb

```diff
@@ -605,7 +605,7 @@ def fact_plan(name = 'facts_test')
           result = run_cli_json(run_command)
           expect(result).not_to include('kind')
           # If puppetdb config loaded this would be fake
-          expect(result['items'][0]['result']['stdout']).to eq("#{conn[:system_user]}\n")
+          expect(result['items'][0]['result']['stdout']).to eq("#{conn[:user]}\n")
```

Comment:
This is an interesting side effect of adding any target to the all group with get_targets. In this case the inventoryfile looks like:
```
---
version: 2
targets:
- _plugin: puppetdb
  query: inventory { facts.fact1 = true }
  config:
    ssh: {}
config:
  transport: ssh
  ssh:
    host: localhost
    user: bolt
    port: 20022
    host-key-check: false
```
previously `run command whoami -t localhost` would use the local transport (the magic of localhost defaults) because there is no `localhost` target in inventory, however now that it is added to the all group it picks up the transport config on the `all` group and the configured ssh transport is used.

Reply:
ugh that seems like a problem

Reply:
I think this is ok if it was previously the case when `--transport` was passed or transport `bolt.yaml`

Reply:
does not appear to be the case:
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt/Boltdir$ cat bolt.yaml
---
transport: winrm
cas@cas-ThinkPad-T460p:~/working_dir/bolt/Boltdir$ bolt command run whoami -t localhost
Started on localhost...
Finished on localhost:
  STDOUT:
    cas
Successful on 1 node: localhost
Ran on 1 node in 0.0 sec
cas@cas-ThinkPad-T460p:~/working_dir/bolt/Boltdir$ bolt command run whoami -t localhost --transport winrm
Started on localhost...
Finished on localhost:
  STDOUT:
    cas
Successful on 1 node: localhost
Ran on 1 node in 0.0 sec
```
Do we need to make `localhost` even more special?

Reply:
(note this is based on master, not the PR branch), though that would be the same with empty inventory 2.